params/networkhelper/provider_utils.go

```diff
@@ -117,19 +118,25 @@ func OverrideEmbeddedProxyProviders(networks []params.Network, enabled bool, use
 	return updatedNetworks
 }
 
-func deepCopyNetworks(networks []params.Network) []params.Network {
+func DeepCopyNetwork(network params.Network) params.Network {
```

Comment:
Perhaps this could be a method of `params.network`?

Reply:
I agree with Igor's suggestion for this and the function below.